src/code/z_eventmgr.c

```diff
@@ -1,52 +1,554 @@
+}
+
+void CutsceneManager_End(void) {
+    ActorCutscene* csEntry;
+    s16 oldCamId;
+    s16 oldStateFlags;
+
+    switch (sCutsceneMgr.startMethod) {
+        case CS_START_2:
+            sCutsceneMgr.targetActor->flags &= ~ACTOR_FLAG_100000;
+            // fallthrough
+        case CS_START_1:
+            func_800B7298(sCutsceneMgr.play, 0, PLAYER_CSMODE_END);
+            sCutsceneMgr.startMethod = CS_START_0;
+            break;
+
+        default:
+            break;
+    }
+
+    csEntry = CutsceneManager_GetCutsceneImpl(sCutsceneMgr.csId);
+
+    switch (csEntry->endSfx) {
+        case CS_END_SFX_TRE_BOX_APPEAR:
+            play_sound(NA_SE_SY_TRE_BOX_APPEAR);
+            break;
+
+        case CS_END_SFX_CORRECT_CHIME:
+            play_sound(NA_SE_SY_CORRECT_CHIME);
+            break;
+
+        default: // CS_END_SFX_NONE
+            break;
+    }
+
+#define RET_CAM sCutsceneMgr.play->cameraPtrs[sCutsceneMgr.retCamId]
+#define CUR_CAM sCutsceneMgr.play->cameraPtrs[sCutsceneMgr.subCamId]
+
+    switch (csEntry->endCam) {
+        case CS_END_CAM_SMOOTH:
+            Play_CopyCamera(sCutsceneMgr.play, sCutsceneMgr.retCamId, sCutsceneMgr.subCamId);
+            RET_CAM->stateFlags =
+                (RET_CAM->stateFlags & ~CAM_STATE_UNDERWATER) | (CUR_CAM->stateFlags & CAM_STATE_UNDERWATER);
+            CutsceneManager_Queue(CS_ID_GLOBAL_RETURN_TO_CAM);
+            break;
+
+        case CS_END_CAM_0:
+        default:
+            Play_CopyCamera(sCutsceneMgr.play, sCutsceneMgr.retCamId, sCutsceneMgr.subCamId);
+            RET_CAM->stateFlags =
+                (RET_CAM->stateFlags & ~CAM_STATE_UNDERWATER) | (CUR_CAM->stateFlags & CAM_STATE_UNDERWATER);
+            break;
+
+        case CS_END_CAM_1:
+            oldCamId = RET_CAM->camId;
+            oldStateFlags = RET_CAM->stateFlags;
+
+            if (sCutsceneMgr.isCameraStored) {
+                // Restore the camera that was stored in subCam 2
+                memcpy(RET_CAM, &sCutsceneMgr.play->subCameras[2], sizeof(Camera));
+
+                RET_CAM->stateFlags =
+                    (RET_CAM->stateFlags & ~CAM_STATE_UNDERWATER) | (CUR_CAM->stateFlags & CAM_STATE_UNDERWATER);
+
+                RET_CAM->stateFlags = (RET_CAM->stateFlags & ~CAM_STATE_2) | (oldStateFlags & CAM_STATE_2);
+                sCutsceneMgr.isCameraStored = false;
+            }
+            RET_CAM->camId = oldCamId;
+            break;
+    }
+
+    if (sCutsceneMgr.subCamId != SUB_CAM_ID_DONE) {
+        Play_ClearCamera(sCutsceneMgr.play, sCutsceneMgr.subCamId);
+        Play_ChangeCameraStatus(sCutsceneMgr.play, sCutsceneMgr.retCamId, CAM_STATUS_ACTIVE);
+    }
+
+    sCutsceneMgr.csId = CS_ID_NONE;
+    sCutsceneMgr.endCsId = CS_ID_NONE;
+    sCutsceneMgr.length = -1;
+    sCutsceneMgr.targetActor = NULL;
+    sCutsceneMgr.subCamId = SUB_CAM_ID_DONE;
+}
+
+s16 CutsceneManager_Update(void) {
+    s16 sp1E = 0;
+
+    if (CutsceneManager_IsNext(CS_ID_GLOBAL_RETURN_TO_CAM) != 0) {
```

Comment:
```suggestion
    if (CutsceneManager_IsNext(CS_ID_GLOBAL_RETURN_TO_CAM)) {
```

Reply:
This function is weird, because it can return other values other than 0 or 1. It can also return -1 and 0x7F. But in practice, it's only ever used to check against `!= 0` (at least with what's been decomped so far). So is it fine to still have a boolean name (`IsNext`) with a boolean treatment (removing `!= 0` as suggested)?

Reply:
Hmm, if `IsNext` is kept I think we should remove the `!= 0`. Not sure what to name it otherwise tbh.

Reply:
`IsNext` seems to be fine for the purpose this function is used for. I've gone and cleaned up this function with all its `!= 0` and `== 0`